aether-ui/aether/ui/api/utils.py

```diff
@@ -216,167 +188,130 @@ def is_object_linked(kernel_refs, object_name, entity_type_name=''):
-        if not is_object_linked(pipeline.kernel_refs, 'projectSchema', entity_type['name']):
-            get_by_name = kernel_data_request('projectschemas/?name={}-{}'.format(
-                                              project_name, entity_type['name']))['results']
-            if len(get_by_name):
-                project_schema_name = '{}-{}'.format(project_name, entity_type['name'])
-                outcome['exists'].append({project_schema_name: 'Project schema {} exists on kernel'.format(
-                                        project_schema_name)})
-    if is_object_linked(pipeline.kernel_refs, 'mapping'):
-        outcome['exists'].append({pipeline.name: 'Mapping with id {} exists'.format(
-                                        pipeline.kernel_refs['mapping'])})
+                outcome['ids']['schema'][entity_type['name']] = get_by_name[0]['id']
+    if is_object_linked(contract.kernel_refs, 'mappings'):
+        outcome['exists'].append({contract.name: 'Mapping with id {} exists'.format(
+                                        contract.kernel_refs['mappings'])})
+        outcome['ids']['mapping'][contract.name] = contract.kernel_refs['mappings']
     else:
-        get_by_name = kernel_data_request(f'mappings/?name={pipeline.name}')['results']
+        get_by_name = kernel_data_request(f'mappings/?name={contract.name}')['results']
         if len(get_by_name):
-            outcome['exists'].append({pipeline.name: 'Pipeline (mapping) with name {} exists on kernel.'.format(
-                                    pipeline.name)})
+            outcome['exists'].append({contract.name: 'Pipeline (mapping) with name {} exists on kernel.'.format(
+                                    contract.name)})
+            outcome['ids']['mapping'][contract.name] = get_by_name[0]['id']
+
+    if pipeline.mappingset:
+        try:
+            mappingset = kernel_data_request(f'mappingsets/{pipeline.mappingset}/')
+            if json.dumps(mappingset['input'], sort_keys=True) != json.dumps(pipeline.schema, sort_keys=True):
+                outcome['exists'].append({pipeline.name: 'Input data will be changed'})
+        except Exception as e:
+            print(str(e))
     return outcome
 
 
-def publish_pipeline(pipeline, projectname, overwrite=False):
+def publish_pipeline(pipeline, projectname, contract, objects_to_overwrite={}):
     '''
-    Transform pipeline to kernel data and publish
+    Transform pipeline and contract to kernel artefacts and publish
     '''
-    outcome = {
-        'successful': [],
-        'error': [],
-        'exists': []
-    }
-    pipeline.kernel_refs = pipeline.kernel_refs if pipeline.kernel_refs else {}
-    project_data = {
-        'revision': str(uuid.uuid4()),
-        'name': projectname,
-        'salad_schema': '[]',
-        'jsonld_context': '[]',
-        'rdf_definition': '[]'
-    }
+    project_id = str(uuid.uuid4())
+    project_name = 'Aux'
```

Comment:
Why is the project name hard coded?

Reply:
Because all pipelines not previously linked to a project have to be identified as coming from AUX, thus an aux project on kernel. And there is no way to specify a project from the ui

Reply:
> And there is no way to specify a project from the ui

But you agree with me that there should be one (not in this PR of course), right?

Reply:
But my doubt goes more to the comment below. I'll try to put all together.

You include the `projectname` as an argument
```python
def publish_pipeline(pipeline, projectname, contract, objects_to_overwrite={}):
```
The next thing you do is to overwrite that value
```python
    project_name = 'Aux'
```
Wait, it's another variable that looks similar :scream: `projectname` is not used at all :woman_shrugging: 

Now with the project id...

Firstly, you set a new value instead of checking `pipeline.kernel_refs['project']` value
```python
    project_id = str(uuid.uuid4())
```
Next, you don't try to find that existing project, you go directly to search the "Aux" project
```python
    projects = kernel_data_request(f'projects/?name=Aux')['results']
```

Do you see my confusion when I review this?

Reply:
> > And there is no way to specify a project from the ui
> 
> But you agree with me that there should be one (not in this PR of course), right?

Yes I do, but it's more of a UX problem than it is code.